### PR TITLE
Add Firebase Extensions Instance resource

### PR DESCRIPTION
**Pull request by rainshen49**
Add support to create a Firebase Extension via Terraform. We are starting with only supporting installing an extension via a published `extension_ref`.

https://github.com/hashicorp/terraform-provider-google/issues/14601

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_firebase_extensions_instance`
```

### CODE REVIEW

File: mmv1/templates/terraform/examples/firebase_extentions_instance_resize_image.tf.erb
Comment (rainshen49) [inline mmv1/templates/terraform/examples/firebase_extentions_instance_resize_image.tf.erb:1]: @pavelgj @tylerg-dev See the example in this file for supported fields. Is there any other field we should add? Is there any field we shouldn't support?

**Comment by SarahFrench** [comment]
The tests above failed due to `Error: Error creating Instance: googleapi: Error 403: Firebase Extensions API has not been used in project ....`

That API should be enabled now so I'll trigger the tests

/gcbrun

File: mmv1/third_party/terraform/tests/resource_firebase_extensions_instance_test.go.erb
Comment (SarahFrench) [inline mmv1/third_party/terraform/tests/resource_firebase_extensions_instance_test.go.erb:128]: From the latest VCR tests it looks like this line needs to be changed to avoid this validation error from the API:

> googleapi: Error 400: Request had 1 validation errors: `params.IMAGE_TYPE`: contains one or more invalid options, each must be one of `jpeg`, `webp`, `png`, `tiff`, `gif`, `avif`, `false`.

Comment (rainshen49) [reply mmv1/third_party/terraform/tests/resource_firebase_extensions_instance_test.go.erb]: Fixed. Please don't run the next test yet. I'm figuring out what's going on with the service account permission error.

File: mmv1/templates/terraform/examples/firebase_extentions_instance_resize_image.tf.erb
Comment (rainshen49) [inline mmv1/templates/terraform/examples/firebase_extentions_instance_resize_image.tf.erb:12]: @pavelgj @johan I'm getting the following permission denied error on CreateExtensionInstance. Is there another permission missing somewhere? Does the Extensions API use a service account and needs to be able to act as this service account just created?

```
generic::permission_denied: Service account `tf-test-sa@<project-id>.iam.gserviceaccount.com` does not exist in `projects/<project-id>` or the caller does not have the correct permissions to use this service account [google.rpc.error_details_ext] { code: 7 message: \"Service account `tf-test-sa@<project-id>.iam.gserviceaccount.com` does not exist in `projects/<project-id>` or the caller does not have the correct permissions to use this service account\"
```

File: mmv1/templates/terraform/examples/firebase_extentions_instance_resize_image.tf.erb
Comment (pavelgj) [inline mmv1/templates/terraform/examples/firebase_extentions_instance_resize_image.tf.erb:29]: Providing a custom `service_account_email` is currently considered a "legacy" feature, which we might deprecate in the future... I'm thinking we should support it as "output only" in TF.

**Review by pavelgj — CHANGES_REQUESTED** [review]

File: mmv1/templates/terraform/examples/firebase_extentions_instance_resize_image.tf.erb
Comment (rainshen49) [reply mmv1/templates/terraform/examples/firebase_extentions_instance_resize_image.tf.erb]: Okay!

**Comment by rainshen49** [comment]
> I'm going to add `service_account_email` to the sample.

Made this an output field instead based on feedback

File: mmv1/templates/terraform/examples/firebase_extentions_instance_resize_image.tf.erb
Comment (rainshen49) [reply mmv1/templates/terraform/examples/firebase_extentions_instance_resize_image.tf.erb:1]: @joehan @pavelgj Can I get one of you to LGTM from the Extensions side if this looks good?

**Review by joehan — APPROVED** [review]
LGTM

**Review by pavelgj — APPROVED** [review]

**Comment by rainshen49** [comment]
@SarahFrench this PR is ready for review.

**Review by SarahFrench — APPROVED** [review]
From reading through the code and the conversations in this PR I'm happy to merge.

For those who may look at this PR in future, this bit of [context about service accounts is useful to know](https://github.com/GoogleCloudPlatform/magic-modules/pull/8133#discussion_r1231524032):

>Providing a custom service_account_email is currently considered a "legacy" feature, which we might deprecate in the future... I'm thinking we should support it as "output only" in TF.

**Pull request merged**
